internal/services/filehash/filesystemhasher.go

```diff
@@ -0,0 +1,107 @@
+package filehash
+
+import (
+	"context"
+	"hash"
+
+	"github.com/ScienceSoft-Inc/integrity-sum/internal/utils/fshasher"
+	"github.com/ScienceSoft-Inc/integrity-sum/pkg/hasher"
+	"github.com/sirupsen/logrus"
+)
+
+type FileHash struct {
+	Path string
+	Hash string
+}
+
+type FileSystemHasher struct {
+	log     *logrus.Logger
+	alg     string
+	workers int
+}
+
+func NewFileSystemHasher(log *logrus.Logger, alg string, workers int) *FileSystemHasher {
+	return &FileSystemHasher{
+		log:     log,
+		alg:     alg,
+		workers: workers,
+	}
+}
+
+// Calculate calculate file hashes synchronously and store into slice
+func (fhs *FileSystemHasher) CalculateAll(ctx context.Context, path string) ([]FileHash, error) {
+	hashChan := make(chan FileHash)
+	result := make([]FileHash, 0, 1024)
+	var err error
+
+	go func() {
+		hashFuncBuilder := fshasher.FileHasherByHash(func() hash.Hash { return hasher.NewHashSum(fhs.alg) })
+		fhs.log.Debug("Begin calculate hashes")
+		err = fshasher.Walk(ctx, fhs.workers, path, hashFuncBuilder, func(path, fhash string) error {
+			hashChan <- FileHash{Path: path, Hash: fhash}
+			return nil
+		})
```

Comment:
1. error is not used

2. Instead of function, you may use a channel directly

Reply:
1. error used outside of goroutine line 51
2. fshasher.Walk run few jobs to calculate hashes, so I aggregate results into slice through channel